connectors/sources/sharepoint.py

```diff
@@ -0,0 +1,715 @@
+                ):
+                    await self.close()
+                    self.session = await self._generate_session()
+                retry += 1
+                if retry > self.retry_count:
+                    raise exception
+                logger.warning(
+                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
+                )
+                await asyncio.sleep(RETRY_INTERVAL**retry)
+
+    def format_document(
+        self,
+        item,
+        document_type,
+        item_type=None,
+        file_name=None,
+    ):
+        """Prepare key mappings for sites, lists, list items and drive items
+
+        Args:
+            item (dictionary): Document from Sharepoint.
+            document_type(string): Type of document(i.e. list_item and drive_item).
+            item_type(string, optional): Type of item i.e. File or Folder. Defaults to None.
+            file_name(string, optional): Name of file. Defaults to None.
+
+        Returns:
+            dictionary: Modified document with the help of adapter schema.
+        """
+        document = {"type": document_type}
+
+        if document_type in [LISTS, DOCUMENT_LIBRARY]:
+            document["url"] = urljoin(
+                self.host_url, item["RootFolder"]["ServerRelativeUrl"]
+            )
+
+        elif document_type == DRIVE_ITEM:
+            document.update(
+                {
+                    "_id": item["GUID"],
+                    "size": item.get("File", {}).get("Length"),
+                    "url": urljoin(self.host_url, item[item_type]["ServerRelativeUrl"]),
+                    "type": item_type,
+                }
+            )
+
+        elif document_type == LIST_ITEM:
+            document.update(
+                {
+                    "_id": item["_id"] if "_id" in item.keys() else item["GUID"],
+                    "file_name": file_name,
+                    "size": item.get("size"),
+                    "url": item["url"],
+                }
+            )
+
+        for elasticsearch_field, sharepoint_field in SCHEMA[document_type].items():
+            document[elasticsearch_field] = (
+                item[item_type][sharepoint_field]
+                if document_type in [DRIVE_ITEM]
+                else item[sharepoint_field]
+            )
+
+        return document
+
+    async def ping(self):
+        """Verify the connection with Sharepoint"""
+
+        self._validate_configuration()
+
+        if not self.ssl_disabled:
+            self.ssl_ctx = self._ssl_context(certificate=self.certificate)
+
+        self.session = await self._generate_session()
+        try:
+            await anext(
+                self._api_call(
+                    url_name=PING,
+                    site_collections=self.site_collections[0],
+                    host_url=self.host_url,
+                )
+            )
+            logger.info("Successfully connected to the Sharepoint")
```

Comment:
```suggestion
            logger.debug("Successfully connected to the Sharepoint")
```

Reply:
Also you can add host_url so that it's easier to figure out where it was connecting